test/test_namedtensor.py

```diff
@@ -272,6 +272,23 @@ def test_diagonal(self):
         self.assertEqual(named_tensor.diagonal(outdim='E', dim1='B', dim2='D').names,
                          ['A', 'C', 'E'])
 
+    def test_max_pooling(self):
+        for device in torch.testing.get_all_device_types():
+            named_tensor_1d = torch.zeros(2, 3, 5, device=device, names=list('ABC'))
+            named_tensor_2d = torch.zeros(2, 3, 5, 7, device=device, names=list('ABCD'))
+            named_tensor_3d = torch.zeros(2, 3, 5, 7, 9, device=device, names=list('ABCDE'))
+
+            self.assertEqual(F.max_pool1d(named_tensor_1d, 2).names, named_tensor_1d.names)
+            self.assertEqual(F.max_pool2d(named_tensor_2d, [2, 2]).names, named_tensor_2d.names)
+            self.assertEqual(F.max_pool3d(named_tensor_3d, [2, 2, 2]).names, named_tensor_3d.names)
+
+            self.assertEqual(F.max_pool1d_with_indices(named_tensor_1d, 2)[0].names, named_tensor_1d.names)
+            self.assertEqual(F.max_pool1d_with_indices(named_tensor_1d, 2)[1].names, named_tensor_1d.names)
```

Comment:
nit: (this isn't blocking) `F.max_pool1d_with_indices(named_tensor_1d, 2)` returns a tuple. We end up computing this twice. Instead, we can compute it once and then check the names on the result:
```
def check_tuple_return(op, inputs, expected_names):
    values, indices = op(*inputs)
    self.assertEqual(values.names, expected_names)
    self.assertEqual(indices.names, expected_names)

check_operator(F.max_pool1d_with_indices, [namedtensor_1d, 2], named_tensor_1d.names)
check_operator(F.max_pool2d_with_indices, [namedtensor_2d, [2, 2]], named_tensor_2d.names)
check_operator(F.max_pool3d_with_indices, [namedtensor_3d, [2, 2, 2]], named_tensor_3d.names)
```

This would improve how fast this test runs by doing less. It shouldn't really matter though so feel free to ship without this.

Reply:
Updated